src/utils/CombineThemes.ts

```diff
@@ -0,0 +1,25 @@
+const isValidProperty = (
+  value: string,
+  themeObject: object,
+): value is keyof typeof themeObject => {
+  return value in themeObject
+}
+const combineThemes = (theme: any, secondTheme: any) => {
+  const combinedTheme = { ...theme }
+  Object.keys(secondTheme).forEach(element => {
+    if (
+      isValidProperty(element, combinedTheme) &&
+      isValidProperty(element, secondTheme)
+    )
+      combinedTheme[element] = {
+        ...combinedTheme[element],
+        ...(secondTheme[element] as object),
+      }
+    else
+      combinedTheme[element] =
+        isValidProperty(element, secondTheme) && secondTheme[element]
+  })
+  return combinedTheme
+}
```

Comment:
there might be a more performant way to do this. 
May be you can see how this works.
https://www.npmjs.com/package/lodash.merge

Reply:
okay bro. I will look after it